fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQueryRenderer.java

```diff
@@ -0,0 +1,2042 @@
+        final String lrAliasName = "LR0";
+
+        // The core data query joining together the logical resources table. Query
+        // parameters are bolted on as exists statements in the WHERE clause. The final
+        // query is constructed when joinResources is called.
+        SelectAdapter select = Select.select("LR0.LOGICAL_RESOURCE_ID", "LR0.LOGICAL_ID", "LR0.CURRENT_RESOURCE_ID");
+        select.from(xxLogicalResources, alias(lrAliasName))
+            .where(lrAliasName, IS_DELETED).eq().literal("N");
+        return new QueryData(select, lrAliasName, null, rootResourceType, 0);
+    }
+
+    @Override
+    public QueryData getParameterBaseQuery(QueryData parent) {
+        final int aliasIndex = getNextAliasIndex();
+        final String xxLogicalResources = parent.getResourceType() + "_LOGICAL_RESOURCES";
+        final String lrAlias = "LR" + aliasIndex;
+        final String parentLRAlias = parent.getLRAlias();
+
+        // SELECT 1 FROM xx_LOGICAL_RESOURCES LRn
+        //    INNER JOIN ...
+        //    INNER JOIN ...
+        //         WHERE LRn.IS_DELETED = 'N'
+        //           AND LRn.LOGICAL_RESOURCE_ID = LRp.LOGICAL_RESOURCE_ID
+        SelectAdapter exists = Select.select("1");
+        exists.from(xxLogicalResources, alias(lrAlias))
+         .where(lrAlias, "IS_DELETED").eq().literal("N") // TODO remove either from here or parent
+         .and(lrAlias, "LOGICAL_RESOURCE_ID").eq(parentLRAlias, "LOGICAL_RESOURCE_ID"); // correlate to parent query
+
+        // Add this exists to the parent query
+        parent.getQuery().from().where().and().exists(exists.build());
+
+        // We are starting a sub-query, so future params need to add themselves to this
+        return new QueryData(exists, lrAlias, null, parent.getResourceType(), 0);
+    }
+
+    @Override
+    public QueryData joinResources(QueryData queryData) {
+        final SelectAdapter logicalResources = queryData.getQuery();
+        final String xxResources = resourceResources(queryData.getResourceType());
+        final String lrAliasName = "LR";
+        SelectAdapter select = Select.select("R.RESOURCE_ID", "R.LOGICAL_RESOURCE_ID", "R.VERSION_ID", "R.LAST_UPDATED", "R.IS_DELETED", "R.DATA", "LR.LOGICAL_ID");
+        select.from(logicalResources.build(), alias(lrAliasName))
+            .innerJoin(xxResources, alias("R"), on(lrAliasName, "CURRENT_RESOURCE_ID").eq("R", "RESOURCE_ID"));
+
+        // The final query (still needs ordering/pagination to be applied
+        return new QueryData(select, lrAliasName, null, queryData.getResourceType(), queryData.getChainDepth());
+    }
+
+    @Override
+    public QueryData includeRoot(String rootResourceType) {
+
+        /* Final query should like this:
+            SELECT RESOURCE_ID, LOGICAL_RESOURCE_ID, VERSION_ID, LAST_UPDATED, IS_DELETED, DATA, LOGICAL_ID
+              FROM (
+                  SELECT DISTINCT R0.RESOURCE_ID, R0.LOGICAL_RESOURCE_ID, R0.VERSION_ID, R0.LAST_UPDATED,
```

Comment:
is distinct not yet eliminated? that's the switch to EXISTS?

Reply:
Old comment. Updated with new query.